contracts/wallet.sol

```diff
@@ -595,12 +595,12 @@ contract Wallet is ENSResolvable, GasTopUpLimit, LoadLimit, AddressWhitelist, Sp
     /// @param _data abi encoded data payload.
     /// @param _signature signed prefix + data.
     function executeRelayedTransaction(uint256 _nonce, bytes calldata _data, bytes calldata _signature) external onlyController {
-        // expecting prefixed data ("rlx:") indicating relayed transaction...
+        // Expecting prefixed data ("monolith:") indicating relayed transaction...
         // ...and an Ethereum Signed Message to protect user from signing an actual Tx
-        bytes32 dataHash = keccak256(abi.encodePacked("rlx:", _nonce, _data)).toEthSignedMessageHash();
-        // verify signature validity i.e. signer == owner
+        bytes32 dataHash = keccak256(abi.encodePacked("monolith:", _nonce, address(this), _data)).toEthSignedMessageHash();
```

Comment:
do we want to put the chainID in here, to make it that much more robust ?

Reply:
@i-stam ?

Reply:
i guess it would be nice, but we have to find a way to verify it, for this type of contract my intuition is that `owner` + `wallet address` are enough

Reply:
as per Slack message - solidity already has a way of obtaining the chain ID of current chain. So we could also sign a chain ID and then obtain it on-chain and check if they match.

Reply:
yeah, since solidity 0.5.12 or something